schematic/schemas/generator.py

```diff
@@ -207,8 +208,10 @@ def get_node_range(self,
             # get node range in the order defined in schema for given node
             required_range = self.se.explore_class(node_label)["range"]
         except KeyError:
-            print("The given source node cannot be found in the graph. Please try a different node label.")
-            return []
+            logger.error(f"The provided source node {node_label} does not exist in the graph. "
+                        "Please use a different node.")
+            raise ValueError(f"The source node {node_label} provided does not "
+                            "exist in the graph.")
```

Comment:
We should avoid debug messages that are essentially duplicated with the error message. In this case, I would just make the error message include everything and drop the `logger` statement.